benches/my_benchmark.rs

```diff
@@ -0,0 +1,196 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+
+use rand::{prelude::IteratorRandom, rngs::OsRng, CryptoRng, Rng, RngCore};
+use std::collections::HashMap;
+use tss_ecdsa::{errors::Result, Identifier, Message, Participant, ParticipantIdentifier};
+
+/// Delivers all messages into their respective participant's inboxes
+fn deliver_all(
+    messages: &[Message],
+    inboxes: &mut HashMap<ParticipantIdentifier, Vec<Message>>,
+) -> Result<()> {
+    for message in messages {
+        for (&id, inbox) in &mut *inboxes {
```

Comment:
I don't really understand how `&mut *inboxes` works. I see the compiler suggests doing it this way :sweat_smile:

Can we do it this way:
```diff
+ for (id, inbox) in inboxes.into_iter() {
+     if *id == message.to() {
- for (&id, inbox) in &mut *inboxes {
-     if id == message.to() {
```
I think it makes it a little more obvious what we are doing.